### PR TITLE
Add shaderc dependency lookup logic

### PR DESCRIPTION
**Pull request by Akaricchi**
> [Dude, like, what even is pkg-config lmfao? XD](https://github.com/google/shaderc/issues/392)
— Google Inc. engineers

### CODE REVIEW

**Comment by dcbaker** [comment]
Akaricchi, before we merge this, let me go do some yelling :)

**Comment by Akaricchi** [comment]
@dcbaker, I would not expect the yelling to be fruitful; the issue has been open since 2017 and they are *still* figuring it out.

**Comment by dcbaker** [comment]
I happen to work in the open-source graphics space and have some connections at Google who do understand how pkg-config works and would have an interest in making this work. 

However, since they're being really slow with this, I think I'm going to go ahead and review this so we can merge it, we can always deprecate it later if we want to.

File: docs/markdown/Dependencies.md [hunk 1]
```diff
@@ -483,6 +483,16 @@ Meson substitutes `modules` to `wx-config` invocation, it generates
 - `compile_args` using `wx-config --cxxflags $modules...`
 - `link_args` using `wx-config --libs $modules...`
 
+## Shaderc
+
+*(added 0.50.1)*
```
Comment (dcbaker) [inline docs/markdown/Dependencies.md]: probably 0.51 at this point.

File: mesonbuild/dependencies/misc.py [hunk 2]
```diff
@@ -666,3 +667,52 @@ def tool_finish_init(ctdep):
...
+            candidates.append(functools.partial(CMakeDependency, 'Shaderc', environment, kwargs))
+
+        if DependencyMethods.SYSTEM in methods:
+            candidates.append(functools.partial(ShadercDependency, environment, kwargs))
+
+        return candidates
+
+    @staticmethod
+    def get_methods():
+        return [DependencyMethods.PKGCONFIG, DependencyMethods.CMAKE, DependencyMethods.SYSTEM]
```
Comment (dcbaker) [inline mesonbuild/dependencies/misc.py]: I'm not sure that it's worth looking for pkg-config and cmake files honestly, it's just going to slow the process down. At the very least let's put SYSTEM first, and if they ever add pkg-config or cmake find modules we can reorder later. In fact, let's just reorder for now.

**Comment by Akaricchi** [comment]
@dcbaker I went ahead and removed the CMake method, but retained the pkg-config one and made system the default. I left the pkg-config in for two reasons: there is at least some talk about supporting it at some point, and people may wish to write their own .pc file that we probably should not break. Also updated the version note and rebased to latest master.

**Comment by dcbaker** [comment]
I agree that leaving pkg-config is the right thing to do, but I think it's also better to use the system approach first since there is no pkg-config implemntation

**Comment by Akaricchi** [comment]
Yes, I did make system the default.

**Pull request merged**